src/Microsoft.Identity.Client/AppConfig/PublicClientApplicationBuilder.cs

```diff
@@ -0,0 +1,101 @@
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+// 
+// ------------------------------------------------------------------------------
+
+using System;
+
+namespace Microsoft.Identity.Client.AppConfig
+{
+    /// <summary>
+    /// </summary>
+    public sealed class PublicClientApplicationBuilder : AbstractApplicationBuilder<PublicClientApplicationBuilder>
+    {
+        /// <inheritdoc />
+        internal PublicClientApplicationBuilder(ApplicationConfiguration configuration)
+            : base(configuration)
+        {
+        }
+
+        /// <summary>
+        /// </summary>
+        /// <param name="options"></param>
+        /// <returns></returns>
+        public static PublicClientApplicationBuilder CreateWithApplicationOptions(PublicClientApplicationOptions options)
+        {
+            var config = new ApplicationConfiguration();
+            return new PublicClientApplicationBuilder(config).WithOptions(options);
+        }
+
+        /// <summary>
+        /// </summary>
+        /// <param name="clientId"></param>
+        public static PublicClientApplicationBuilder Create(string clientId)
+        {
+            var config = new ApplicationConfiguration();
+            return new PublicClientApplicationBuilder(config).WithClientId(clientId);
+        }
+
+        /// <summary>
+        /// </summary>
+        /// <param name="clientId"></param>
+        /// <param name="authority"></param>
+        /// <param name="validateAuthority"></param>
+        /// <returns></returns>
+        public static PublicClientApplicationBuilder Create(string clientId, string authority, bool validateAuthority = true)
+        {
+            return Create(clientId).WithAuthority(authority, validateAuthority, true);
+        }
+
+        /// <summary>
+        /// </summary>
+        /// <param name="tokenCache"></param>
+        /// <returns></returns>
+        public PublicClientApplicationBuilder WithUserTokenCache(ITokenCache tokenCache)
+        {
+            Config.UserTokenCache = tokenCache;
+            return this;
+        }
+
+        /// <summary>
+        /// </summary>
+        /// <returns></returns>
+        public IPublicClientApplication Build()
+        {
+            return BuildConcrete();
+        }
+
+        /// <summary>
+        ///     TODO: do we want this public?  I'd prefer to keep it internal and have the interface be our customer boundary.
+        ///     But some of our apps without InternalsVisibleTo talk to PCA for now.  Need to resolve this.
+        /// </summary>
+        /// <returns></returns>
+        public PublicClientApplication BuildConcrete()
+        {
+            throw new NotImplementedException();
+            // return new PublicClientApplication(BuildConfiguration());
```

Comment:
This will get wired up once we finalize the API and start hooking it into the actual PCA/CCA instances in a later PR.